dawn/src/driver-includes/cuda_verify.cpp

```diff
@@ -0,0 +1,108 @@
+#include "cuda_verify.hpp"
+
+namespace dawn {
+__global__ void isclose_kernel(const int num_el, const double* __restrict__ dsl,
+                               const double* __restrict__ fortran, bool* __restrict__ verified,
+                               const double rel_tol, const double abs_tol) {
+  unsigned int pidx = blockIdx.x * blockDim.x + threadIdx.x;
+  if(pidx >= num_el) {
+    return;
+  }
+
+  // This verification method is inspired by numpy.isclose from Python
+  verified[pidx] = (abs(dsl[pidx] - fortran[pidx]) <= (abs_tol + rel_tol * abs(fortran[pidx])));
+}
+
+bool verify_field(const int num_el, const double* dsl, const double* actual, std::string name,
+                  const double rel_tol, const double abs_tol) {
+  double relErr, absErr;
+  double* gpu_error;
+
+  const int blockSize = 16;
+  dim3 dG((num_el + blockSize - 1) / blockSize);
+  dim3 dB(blockSize);
+
+  gpuErrchk(cudaMalloc((void**)&gpu_error, num_el * sizeof(double)));
+  gpuErrchk(cudaPeekAtLastError());
+  gpuErrchk(cudaDeviceSynchronize());
+
+  compare_kernel<RelErrTag><<<dG, dB>>>(num_el, dsl, actual, gpu_error);
+  gpuErrchk(cudaPeekAtLastError());
+
+  thrust::device_ptr<double> dev_ptr;
+  dev_ptr = thrust::device_pointer_cast(gpu_error);
+  relErr = thrust::reduce(dev_ptr, dev_ptr + num_el, 0., thrust::maximum<double>());
+
+  gpuErrchk(cudaPeekAtLastError());
+  std::cout << "[DSL] " << name << " relative error: " << std::scientific << relErr << "\n"
+            << std::flush;
+
+  compare_kernel<AbsErrTag><<<dG, dB>>>(num_el, dsl, actual, gpu_error);
+  gpuErrchk(cudaPeekAtLastError());
+
+  dev_ptr = thrust::device_pointer_cast(gpu_error);
+  absErr = thrust::reduce(dev_ptr, dev_ptr + num_el, 0., thrust::maximum<double>());
+  gpuErrchk(cudaPeekAtLastError());
+
+  std::cout << "[DSL] " << name << " absolute error: " << std::scientific << absErr << "\n"
+            << std::flush;
+  gpuErrchk(cudaFree(gpu_error));
+  gpuErrchk(cudaPeekAtLastError());
```

Comment:
I like the overview of the dataset printed by the code below.
I'd also give an overview of the errors here, i.e. print always both the max (currently printed) and min relative and absolute errors. Not sure about the avg.
Providing more infos here helps in reducing the number of times we need to download the vtk/csv files and visualize them.